### PR TITLE
chore: refactoring opacity tokens and removing and adding some colors

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request refactors the opacity token to be outside of the `component-library` folder, as it is used outside of the component library. Additionally, it removes some deprecated colors and re-adds missing overlay colors.

**Reason for the change:**
The opacity token is utilized beyond the scope of the `component-library`, so it needs to be accessible outside of this folder for better maintainability and usage consistency. Furthermore, removing deprecated colors and re-adding missing overlay colors ensures the codebase is clean and up-to-date with the latest design requirements.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25158)

## **Related issues**

Part of: https://github.com/MetaMask/metamask-extension/issues/24916

## **Manual testing steps**

1. Navigate to the Button component in storybook.
2. Check `disabled` control and check opacity token still works
3. Check codebase for instances of `success-alternative`, `infoDisabled`, and `info-disabled` to ensure none remain

## **Screenshots/Recordings**

Before after screenshots in the code comments for easier review

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability.
- [x] I’ve included tests if applicable.
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/component-library/component-library-components.scss [hunk 1]
```diff
@@ -5,11 +5,6 @@
 * unintended overrides.
 **/
 
-// Temp CSS Variables
-:root {
-  --opacity-disabled: 0.5;
-}
-
```
Comment (georgewrmarshall) [inline ui/components/component-library/component-library-components.scss:12]: Moved to `opacity.scss` as it's used outside of the component-library components now https://github.com/MetaMask/metamask-extension/pull/25158/files#diff-0bee63110d632972bcb15aeea3a018010055fb4c48040c989dfc7dbc329e4876

File: ui/helpers/constants/design-system.ts [hunk 2]
```diff
@@ -35,7 +35,6 @@ export enum Color {
   infoDefault = 'info-default',
   infoMuted = 'info-muted',
   infoInverse = 'info-inverse',
-  infoDisabled = 'info-disabled',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:38]: Must have been a bad merge but `infoDisabled` no longer exists in the codebase

Screenshot 2024-06-07 at 4 03 35 PM

File: ui/css/utilities/opacity.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Temp CSS Variables
+:root {
+  --opacity-disabled: 0.5;
+}
```
Comment (georgewrmarshall) [inline ui/css/utilities/opacity.scss:4]: Opacity token still work as expected after refactoring

Screenshot 2024-06-07 at 4 04 44 PM

File: ui/css/index.scss [hunk 4]
```diff
@@ -5,6 +5,7 @@
 @import './utilities/fonts.scss';
 @import './utilities/pending-colors.scss';
 @import './utilities/colors.scss';
+@import './utilities/opacity.scss';
```
Comment (georgewrmarshall) [inline ui/css/index.scss:8]: Importing opacity stylesheet in `index.css`

File: ui/css/design-system/_colors.scss [hunk 5]
```diff
@@ -24,13 +24,11 @@ $color-map: (
   'warning-muted': --color-warning-muted,
   'warning-inverse': --color-warning-inverse,
   'success-default': --color-success-default,
-  'success-alternative': --color-success-alternative,
   'success-muted': --color-success-muted,
   'success-inverse': --color-success-inverse,
   'info-default': --color-info-default,
   'info-muted': --color-info-muted,
   'info-inverse': --color-info-inverse,
-  'info-disabled': --color-info-disabled,
```
Comment (georgewrmarshall) [inline ui/css/design-system/_colors.scss:33]: Must have been a bad merge `info-disabled` no longer exists in the codebase

Screenshot 2024-06-07 at 4 06 04 PM

File: ui/css/design-system/_colors.scss [hunk 6]
```diff
@@ -24,13 +24,11 @@ $color-map: (
   'warning-muted': --color-warning-muted,
   'warning-inverse': --color-warning-inverse,
   'success-default': --color-success-default,
-  'success-alternative': --color-success-alternative,
```
Comment (georgewrmarshall) [inline ui/css/design-system/_colors.scss:27]: Must have been a bad merge `success-alternative` no longer exists in the codebase

Screenshot 2024-06-07 at 4 06 48 PM

File: .storybook/3.COLORS.stories.mdx [hunk 7]
```diff
@@ -106,15 +106,13 @@ var(--color-warning-inverse)
...
-var(--color-success-alternative) [DEPRECATED]
 var(--color-success-muted)
 var(--color-success-inverse)
 
 /** Info */
 var(--color-info-default)
 var(--color-info-muted)
 var(--color-info-inverse)
-var(--color-info-disabled) [DEPRECATED]
 ```
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:116]: Updating docs to remove deprecated CSS variables

File: .storybook/3.COLORS.stories.mdx [hunk 8]
```diff
@@ -74,7 +74,8 @@ var(--color-border-muted)
 
 /** Overlays */
 var(--color-overlay-default)
-var(--color-overlay-inverse) [DEPRECATED]
+var(--color-overlay-alternative)
+var(--color-overlay-inverse)
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:78]: Un-deprecating overlay/inverse as there are [use cases](https://consensys.slack.com/archives/C02VD8QG0LT/p1717083547070679) and adding overlay-alternative. See stylesheet in this branch

Screenshot 2024-06-09 at 8 08 00 PM

File: .storybook/3.COLORS.stories.mdx [hunk 9]
```diff
@@ -84,37 +85,35 @@ var(--color-primary-alternative)
 var(--color-primary-muted)
 var(--color-primary-inverse)
 
-var(--color-secondary-default) [DEPRECATED]
-var(--color-secondary-alternative) [DEPRECATED]
-var(--color-secondary-muted) [DEPRECATED]
-var(--color-secondary-inverse) [DEPRECATED]
-var(--color-secondary-disabled) [DEPRECATED]
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:91]: Removing deprecated secondary colors which will be removed in v4, may need to rebase if this is merged first https://github.com/MetaMask/metamask-extension/pull/24971

File: .storybook/3.COLORS.stories.mdx [hunk 10]
```diff
@@ -84,37 +85,35 @@ var(--color-primary-alternative)
...
-
 /** States */
 /** Error */
 var(--color-error-default)
 var(--color-error-default-hover)
 var(--color-error-default-pressed)
 var(--color-error-alternative)
+var(--color-error-default-hover)
+var(--color-error-default-pressed)
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:95]: These CSS variables are present and allowed to be used. Updating the docs to reflect this.

Screenshot 2024-06-09 at 8 14 35 PM

File: .storybook/3.COLORS.stories.mdx [hunk 11]
```diff
@@ -84,37 +85,35 @@ var(--color-primary-alternative)
...
 var(--color-error-muted)
 var(--color-error-inverse)
 
 /** Warning */
 var(--color-warning-default)
+var(--color-warning-default-hover)
+var(--color-warning-default-pressed)
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:102]: These CSS variables are present and allowed to be used. Updating the docs to reflect this.

Screenshot 2024-06-09 at 8 15 45 PM

File: .storybook/3.COLORS.stories.mdx [hunk 12]
```diff
@@ -84,37 +85,35 @@ var(--color-primary-alternative)
...
 var(--color-warning-muted)
 var(--color-warning-inverse)
 
 /** Success */
 var(--color-success-default)
-var(--color-success-alternative) [DEPRECATED]
+var(--color-success-default-hover)
+var(--color-success-default-pressed)
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:109]: These CSS variables are present and allowed to be used. Updating the docs to reflect this.

Screenshot 2024-06-09 at 8 16 33 PM

File: ui/css/utilities/colors.scss [hunk 13]
```diff
@@ -9,6 +9,10 @@ Before adding a color here make sure that there isn't a design token available.
   --mainnet: #29b6af;
   --inherit: inherit;
   --transparent: transparent;
+  --color-network-goerli-default: #1098fc;
+  --color-network-sepolia-default: #cfb5f0;
+  --color-network-goerli-inverse: #fcfcfc;
+  --color-network-sepolia-inverse: #fcfcfc;
```
Comment (georgewrmarshall) [inline ui/css/utilities/colors.scss:15]: These network colors are slated for removal from v4 of the design tokens but are still used in the codebase. We are moving them to the extension stylesheet as they are third-party colors and will not change. We should deprecate third-party network colors in favor of using images instead.

Comment (georgewrmarshall) [reply ui/css/utilities/colors.scss]: Moved to this PR https://github.com/MetaMask/metamask-extension/pull/24953/files#diff-e94da7b5420af178ae8a2ae327e1dc681421106b5b02b5a1a5f88c959b2bcedb

**Review by brianacnguyen — APPROVED** [review]

**Review by bergeron — APPROVED** [review]

**Pull request merged**